app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,20 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+
+    respond_to do |format|
+      if @scenario.agents.update_all disabled: params[:scenario][:disabled]
```

Comment:
That looks totally fine to me, in the rails console you need to call `@scenario.agents(true)` or `@scenario.reload; @scenario.agents` to fetch the updated data from the database.
The log output also shows that sql `UPDATE` call was successful, are the agents not shown as disabled in the UI?